C error when generic function with none

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 22.04 LTS (WSL 2)
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 3800X 8-Core Processor
CC version: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

getwd: /home/zakuro/src/github.com/cotowali/cotowali
vmodules: /home/zakuro/.vmodules
vroot: /home/zakuro/src/github.com/zakuro9715/v
vexe: /home/zakuro/src/github.com/zakuro9715/v/v
vexe mtime: 2023-01-19 23:03:34
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 bb705c0.fed6524
env VFLAGS: "-cc clang"

Git version: git version 2.34.1
Git vroot status: weekly.2023.03-29-gfed65248 (6 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
fn f[T](v T) {
	println(v)
}

f(none)

```

**What did you expect to see?**

no error


**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_1000/main.13009416327711408029.tmp.c:12176:17: error: passing 'IError' (aka 'struct IError') to parameter of incompatible type 'none' (aka 'struct none')
        main__f_T_none(_const_none__);
                       ^~~~~~~~~~~~~
/tmp/v_1000/main.13009416327711408029.tmp.c:12171:42: note: passing argument to parameter 'v' here
VV_LOCAL_SYMBOL void main__f_T_none(none v) {
                                         ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1019,6 +1019,9 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 			}
 		}
 		arg_typ_sym := c.table.sym(arg_typ)
+		if arg_typ_sym.kind == .none_ {
+			c.error('cannot use `none` as function argument', call_arg.pos)
+		}
 		param_typ_sym := c.table.sym(param.typ)
 		if func.is_variadic && arg_typ.has_flag(.variadic) && node.args.len - 1 > i {
 			c.error('when forwarding a variadic variable, it must be the final argument',
@@ -1747,6 +1750,9 @@ fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 						1} to `${method_name}`', arg.pos)
 				}
 				continue
+			}
+			if final_arg_sym.kind == .none_ {
+				c.error('cannot use `none` as method argument', arg.pos)
 			}
 			if param.typ.is_ptr() && !arg.typ.is_real_pointer() && arg.expr.is_literal()
 				&& !c.pref.translated {

--- a/vlib/v/checker/tests/fn_call_using_none_arg_err.out
+++ b/vlib/v/checker/tests/fn_call_using_none_arg_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/fn_call_using_none_arg_err.vv:6:4: error: cannot use `none` as function argument
+    4 |
+    5 | fn main() {
+    6 |     f(none)
+      |       ~~~~
+    7 | }

--- a/vlib/v/checker/tests/fn_call_using_none_arg_err.vv
+++ b/vlib/v/checker/tests/fn_call_using_none_arg_err.vv
@@ -1,0 +1,7 @@
+fn f[T](v T) {
+	println(v)
+}
+
+fn main() {
+	f(none)
+}

